src/main/java/com/google/codeu/data/EncryptPassword.java

```diff
@@ -0,0 +1,78 @@
+package test;
+
+import java.math.BigInteger;
+import java.util.Random;
+
+//help from samfoundary.com
+
+public class EncryptPassword {
+	
+	public String password;
+	private byte[] stringbytes;
+	private byte[] encryptedpwd;
+	private Random random;
+	private BigInteger publickey;
+	private BigInteger privatekey;
+	private BigInteger prime1; //p
+	private BigInteger prime2; //q
+	private BigInteger exponent;
+	private BigInteger phi; 
+	
+	
+	/**
+	 * constructor
+	 * @param password (the user's original unencrypted password)
+	 */
+	public EncryptPassword(String password) {
+		this.password = password;
+		this.stringbytes = this.password.getBytes();
+		random = new Random();
+		prime1 = BigInteger.probablePrime(1024, random);
+		prime2 = BigInteger.probablePrime(1024, random);
+		publickey = prime1.multiply(prime2); //n = p*q
+		exponent = BigInteger.probablePrime(256, random);
+		phi = prime1.subtract(BigInteger.ONE).multiply(prime2.subtract(BigInteger.ONE));
+		
+		//how we generate the private key

```

Comment:
nice comment here, although I don't understand these lines, thanks to the clear comments I can tell what they are doing